core/includes/date.inc

```diff
@@ -710,7 +724,7 @@ function date_format_date($date, $type = 'medium', $format = '', $langcode = NUL
 /**
  * Formats a time interval with granularity, including past and future context.
  *
- * @param object $date
+ * @param BackdropDateTime|null $date
```

Comment:
Similar to above, can it be null, anyway? The code from 2015 checks for empty, yes. But maybe it was just defensive coding...

Reply:
There's nothing in core or contrib that depends on `date_format_interval()` accepting null so we could probably even make it `@param BackdropDateTime $date`.